File: ce/snippets/csharp/CRMV8/strongtypes/cs/useraccess4.cs
// <snippetuseraccess4>


 var principalAccessReq = new RetrievePrincipalAccessRequest
 {
     Principal = principal,
     Target = leadReference
 };
 var principalAccessRes = (RetrievePrincipalAccessResponse)
     _serviceProxy.Execute(principalAccessReq);
 Console.WriteLine("Access rights of {0} ({1}) on the lead: {2}\r\n",
     GetEntityReferenceString(principal),
     additionalIdentifier,
     principalAccessRes.AccessRights);

// </snippetuseraccess4>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/strongtypesvb/vb/useraccess4.vb
' <snippetuseraccess4>


 Dim principalAccessReq = New RetrievePrincipalAccessRequest With
                          {
                              .Principal = principal,
                              .Target = leadReference
                          }
 Dim principalAccessRes = CType(_serviceProxy.Execute(principalAccessReq), 
     RetrievePrincipalAccessResponse)
 Console.WriteLine("Access rights of {0} ({1}) on the lead: {2}" &amp; vbCrLf,
                   GetEntityReferenceString(principal),
                   additionalIdentifier,
                   principalAccessRes.AccessRights)

' </snippetuseraccess4>